sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1697,6 +1697,8 @@ class Analyzer(
       // Only a few unary nodes (Project/Filter/Aggregate) can contain subqueries.
       case q: UnaryNode if q.childrenResolved =>
         resolveSubQueries(q, q.children)
+      case j: Join if j.childrenResolved =>
+        resolveSubQueries(j, Seq(j, j.left, j.right))
```

Comment:
Can't recall the details, but why it's not `Seq(j.left, j.right)`?

Reply:
> Can't recall the details, but why it's not `Seq(j.left, j.right)`?

Should be a mistake, raise a pr and remove this?